File: seed/csharp-model/trace/src/Client/V2/Problem/CustomFiles.cs
using Client.V2;
using System.Text.Json.Serialization;
using StringEnum;
using Client;

namespace Client.V2;

public class CustomFiles
{
    public class _Basic : BasicCustomFiles
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "basic";
    }
    public class _Custom
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "custom";

        [JsonPropertyName("value")]
        public Dictionary<StringEnum<Language>, Files> Value { get; init; }
    }
}

File: seed/rust-sdk/trace/src/types/custom_files.rs
use crate::basic_custom_files::BasicCustomFiles;
use std::collections::HashMap;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(tag = "type")]
pub enum CustomFiles {
        Basic {
            #[serde(flatten)]
            data: BasicCustomFiles,
        },

        Custom {
            value: HashMap<Language, Files>,
        },
}
